infobip-typescript-generator-extension-api/src/main/java/com/infobip/typescript/validation/CustomAnnotationToTsDecoratorConverter.java

```diff
@@ -0,0 +1,102 @@
+package com.infobip.typescript.validation;
+
+import com.infobip.typescript.CustomTSDecorator;
+import com.infobip.typescript.DecoratorParameterListExtractor;
+import com.infobip.typescript.validation.converter.ObjectToTSLiteralConverter;
+import com.infobip.typescript.validation.exception.CanNotResolveMethod;
+import com.infobip.typescript.validation.exception.TSParameterExtractorInstantiationException;
+import cz.habarta.typescript.generator.emitter.*;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class CustomAnnotationToTsDecoratorConverter {
+
+    private static final String MESSAGE_METHOD = "message";
+
+    private final ValidationMessageReferenceResolver validationMessageReferenceResolver;
+    private final ObjectToTSLiteralConverter objectToTSLiteralConverter;
+
+    CustomAnnotationToTsDecoratorConverter(ValidationMessageReferenceResolver validationMessageReferenceResolver) {
+        this.validationMessageReferenceResolver = validationMessageReferenceResolver;
+        this.objectToTSLiteralConverter = new ObjectToTSLiteralConverter();
+    }
+
+    public Stream<TsDecorator> convert(Annotation annotation) {
+        CustomTSDecorator customTSDecorator = getCustomTSDecorator(annotation);
+        String annotationName = getAnnotationName(annotation, customTSDecorator);
+        Optional<String> message = extractMessage(annotation);
+        Stream<TsExpression> referenceStream = getReferences(annotation, customTSDecorator);
+        return message.isPresent()
+                ? convert(annotation, annotationName, message.get(), referenceStream)
+                : convert(annotationName, referenceStream);
+    }
+
+    private CustomTSDecorator getCustomTSDecorator(Annotation annotation) {
+        return annotation.annotationType().getAnnotation(CustomTSDecorator.class);
+    }
+
+    private String getAnnotationName(Annotation annotation, CustomTSDecorator customTSDecorator) {
+        return customTSDecorator.typeScriptDecorator().isEmpty()
+                ? annotation.annotationType().getSimpleName()
+                : customTSDecorator.typeScriptDecorator();
+    }
+
+    private Stream<TsExpression> getReferences(Annotation annotation, CustomTSDecorator customTSDecorator) {
+        try {
+            DecoratorParameterListExtractor decoratorParameterListExtractor = customTSDecorator.decoratorParameterListExtractor()
+                                                                                               .newInstance();
```

Comment:
deprecated call
see if proposed fix (in IDEA) works for our case

Reply:
could you just tell me what is deprecated, because locally IDE-a does not show me any deprecated method.
I am running this code on java 8 (all project is written on java 8), so i just want to check if it is java version issue or my IDE-a has some buggy behavior :)